examples/platform/silabs/efr32/rs911x/rsi_if.c

```diff
@@ -195,10 +201,41 @@ static void wfx_rsi_join_cb(uint16_t status, const uint8_t * buf, const uint16_t
          * We should enable retry.. (Need config variable for this)
          */
         WFX_RSI_LOG("%s: failed. retry: %d", __func__, wfx_rsi.join_retries);
-#if (WFX_RSI_CONFIG_MAX_JOIN != 0)
-        if (++wfx_rsi.join_retries < WFX_RSI_CONFIG_MAX_JOIN)
-#endif
+        if (!is_disconnection_event)
         {
+            /* After the reboot or a commissioning time device failed to connect with AP.
+             * Device will retry to connect with AP upto WFX_RSI_CONFIG_MAX_JOIN retries.
+             */
+            if (wfx_rsi.join_retries < WFX_RSI_CONFIG_MAX_JOIN)
+            {
+                WFX_RSI_LOG("%s: Next attempt after %d Seconds", __func__, CONVERT_MS_TO_SEC(WLAN_RETRY_TIMER_MS));
+                vTaskDelay(pdMS_TO_TICKS(WLAN_RETRY_TIMER_MS));
+                xEventGroupSetBits(wfx_rsi.events, WFX_EVT_STA_START_JOIN);
+            }
+            else
+            {
+                WFX_RSI_LOG("Connect failed after %d tries", wfx_rsi.join_retries);
+            }
+        }
+        else
+        {
+            /* After disconnection
+             * At the telescopic time interval device try to reconnect with AP, upto WLAN_MAX_RETRY_TIMER_MS intervals
+             * are telescopic. If interval exceed WLAN_MAX_RETRY_TIMER_MS then it will try to reconnect at
+             * WLAN_MAX_RETRY_TIMER_MS intervals.
+             */
+            if (retryInterval < WLAN_MAX_RETRY_TIMER_MS)
```

Comment:
Better logic can be implemented if we are going for a telescopic timer followed by a flat timer.

Reply:
```c
if (retryInterval > WLAN_MAX_RETRY_TIMER_MS) 
{
  retryInterval = WLAN_MAX_RETRY_TIMER_MS;
  // add logs..
} else {
  // add logs..
}
vTaskDelay(pdMS_TO_TICKS(retryInterval));
retryInterval += retryInterval;
```

// add logs where needed.

Reply:
Thanks for your input, optimized the logic for telescopic timer.